vhdl_lang/src/syntax/test.rs

```diff
@@ -39,6 +42,31 @@ pub struct CodeBuilder {
     pub symbols: Arc<Symbols>,
 }
 
+impl AnyDesignUnit {
+    pub fn entity(&self) -> &EntityDeclaration {
+        match self {
+            AnyDesignUnit::Primary(AnyPrimaryUnit::Entity(ent)) => ent,
+            _ => unreachable!("Expected entity"),
```

Comment:
I think panic! is more appropriate here. Unreachable is only used if it can be known locally. Here it is up to the caller to ensure we never call this on the from `DesignUnit`.

I would also have named the function `expect_entity` if it panics on non-entites. Otherwise I would have returned an `Option` and handled it in the caller.